packages/plugin-mapbox/src/main/java/com/openmobilehub/android/maps/plugin/mapbox/presentation/maps/OmhMarkerImpl.kt

```diff
@@ -439,9 +448,11 @@ internal class OmhMarkerImpl(
                 throw IllegalStateException("Failed to remove GeoJsonSource from map: $error")
             }
 
-            val removeImageResult = style.removeStyleImage(getMarkerIconID(isCustomIconSet))
-            removeImageResult.error?.let { error ->
-                throw IllegalStateException("Failed to remove image from map: $error")
+            if (lastMarkerIconID != null) {
+                val removeImageResult = style.removeStyleImage(lastMarkerIconID!!)
```

Comment:
```suggestion
            lastMarkerIconID?.let {
                val removeImageResult = style.removeStyleImage(it)
```

Reply:
Refactored in 3203a66940400da0906cdf14482f0856d617ee7a